drivers: mpu9250: magnetometer data read fix

LorisSchmit opened the pull request:
The magnetometer data read from AK9863 is assumed to be big endian 
(like the accelerometer & gyroscope data).
However it is in little endian, which means the appropriate byteorder
function is sys_le16_to_cpu().
See the MPU925 register map: https://3cfeqx1hf82y3xcoull08ihx-wpengine.netdna-ssl.com/wp-content/uploads/2017/11/RM-MPU-9250A-00-v1.6.pdf
At 5.6 HXL to HZH: Measurement Data

Signed-off-by: Loris Schmit loris.schmit@inovex.de

MaureenHelm reviewed (commented):
Thanks for the fix.

Please fix the compliance failure:
```
Error: See https://docs.zephyrproject.org/latest/contribute/#commit-guidelines for more details

1a268bbab1b15f489f5509b07e84b6eab1168809: author email (lschmit <loris.schmit@inovex.de>) needs to match one of the signed-off-by entries.
1a268bbab1b15f489f5509b07e84b6eab1168809: author email (lschmit <loris.schmit@inovex.de>) does not follow the syntax: First Last <email>.
Error: See https://docs.zephyrproject.org/latest/contribute/#commit-guidelines for more details

1: UC2 Signed-off-by: must have a full name
3: UC4 Line exceeds max length (210>75): "The magnetometer data read from AK9863 is assumed to be big endian (like the accelerometer & gyroscope data). However it is in little endian, which means the appropriate byteorder function is sys_le16_to_cpu()."
```

MaureenHelm commented:
cc: @dchat-nordic

dchat-nordic commented:
I can confirm - the data is actually little endian.

MaureenHelm commented:
> Please fix the compliance failure:
> 
> ```
> Error: See https://docs.zephyrproject.org/latest/contribute/#commit-guidelines for more details
> 
> 1a268bbab1b15f489f5509b07e84b6eab1168809: author email (lschmit <loris.schmit@inovex.de>) needs to match one of the signed-off-by entries.
> 1a268bbab1b15f489f5509b07e84b6eab1168809: author email (lschmit <loris.schmit@inovex.de>) does not follow the syntax: First Last <email>.
> Error: See https://docs.zephyrproject.org/latest/contribute/#commit-guidelines for more details
> 
> 1: UC2 Signed-off-by: must have a full name
> 3: UC4 Line exceeds max length (210>75): "The magnetometer data read from AK9863 is assumed to be big endian (like the a
> ```

This still needs to be fixed

Commit:
drivers: mpu9250: magnetometer data read fix

The magnetometer data read from AK9863 is assumed to be big endian
(like the accelerometer & gyroscope data).
However it is in little endian, which means the appropriate byteorder
function is sys_le16_to_cpu().
See the MPU925 register map:
https://3cfeqx1hf82y3xcoull08ihx-wpengine.netdna-ssl.com/wp-content/uploads/2017/11/RM-MPU-9250A-00-v1.6.pdf
At 5.6 HXL to HZH: Measurement Data

Signed-off-by: Loris Schmit <loris.schmit@inovex.de>

LorisSchmit commented:
Hi, I fixed the compliance failure. However the compliance checks and license scan don't run. Could please check up on that?

LorisSchmit commented:
Hi, I believe I finally got it right. Please run the compliance checks again. Sorry for the trouble, this is my first contribution.

MaureenHelm commented:
@gmarull @teburd @avisconti will one of you please take a look?

teburd commented:
Nice fix
